Modules/Core/Common/test/itkXMLFileOutputWindowTest.cxx

```diff
@@ -46,6 +56,33 @@ itkXMLFileOutputWindowTest(int argc, char * argv[])
 
   logger->SetInstance(logger);
 
+  // If not input filename is provided, remove the contents in the existing file to avoid counting existing lines when
+  // contents are appended
+  if (argc == 1)
+  {
+    // In order to initialize the filename, some text needs to be written first
+    const char * regularText = "text";
+    logger->DisplayText(regularText);
+
+    // Get the filename
+    const char *      fileBaseName = logger->GetFileName();
+    const std::size_t size = 4096;
+    char              tmp[size];
+    char *            status = cwd(tmp, size);
+    if (!status)
+    {
+      std::cerr << "Test failed!" << std::endl;
+      std::cerr << "Error getting the current directory. Cannot delete the contents of the target file: "
+                << fileBaseName << std::endl;
+      return EXIT_FAILURE;
+    }
+    std::string fileName = tmp + std::string(OS_SEP) + std::string(fileBaseName);
+
+    // Delete the contents
+    std::ofstream ofs;
+    ofs.open(fileName, std::ofstream::out | std::ofstream::trunc);
+    ofs.close();
```

Comment:
I could not remove the file using `std::remove`. I ignore why. Please feel free to propose changes if you happen to know how deleting a file is effectively done.

Reply:
[std::remove](https://en.cppreference.com/w/cpp/algorithm/remove) deals with iterators, not files.

Reply:
:roll_eyes: I'd dare to say that it deals with files when the appropriate header is included (which was the case when I tried): https://en.cppreference.com/w/cpp/io/c/remove